src/algebra/module/submodule/pointwise.lean

```diff
@@ -136,6 +136,22 @@ instance pointwise_add_comm_monoid : add_comm_monoid (submodule R M) :=
 @[simp] lemma add_eq_sup (p q : submodule R M) : p + q = p ⊔ q := rfl
 @[simp] lemma zero_eq_bot : (0 : submodule R M) = ⊥ := rfl
 
+/-- This is not an instance, as it would form a simp loop between `bot_eq_zero` and
+`submodule.zero_eq_bot`. It can be safely enabled with
+```lean
+local attribute [-simp] submodule.zero_eq_bot
+local attribute [instance] canonically_ordered_add_monoid
+```
+-/
+def canonically_ordered_add_monoid : canonically_ordered_add_monoid (submodule R M) :=
```

Comment:
make reducible (see reducible non-instance library note)

Reply:
I'm not sure I agree; `@[reduclble] foo` is for when we have `instance : X := foo`, not for when we have `local attribute [instance] foo`.

Reply:
At any rate, the follow-up PR will make this moot.